src/triage/component/architect/feature_dictionary.py

```diff
@@ -0,0 +1,64 @@
+from triage.component.architect.utils import remove_schema_from_table_name
+from triage.util.structs import FeatureNameList
+from collections.abc import Iterable
+from collections import MutableMapping
+
+
+class FeatureDictionary(MutableMapping):
+    """A feature dictionary, consisting of table names as keys and column names as values
+
+    If a list of feature_blocks is passed, will initialize the feature dictionary with their data.
+    """
+    def __init__(self, feature_blocks=None, *args, **kwargs):
+        self.tables = dict()
```

Comment:
Curious, is there a reason to inherit `MutableMapping` rather than simply `dict`? This seems to be just hiding an actual `dict`, with minor adjustments.

And it's a little confusing -- when you `__getitem__` it's a `FeatureNameList`, but only then. It seems like it could be simpler.

```
class FeatureDict(dict):

    def __init__(self, feature_blocks=None, *args, **kwargs):
        super().__init__(*args, **kwargs)   # or super().__init__() and update() if you're trying to avoid __setitem__?

        ...  # existing processing of feature_blocks

    def __setitem__(self, table, feature_names):
        ...  # existing, but only "fair" to accept value which is already a FeatureList
        # and why not set it eagerly?
        self[table] = feature_names if isinstance(feature_names, FeatureNameList) else FeatureNameList(feature_names)

    # no need for __getitem__, __iter__, __len__, __delitem__

Reply:
I mostly haven't subclassed dict because of this article: http://www.kr41.net/2016/03-23-dont_inherit_python_builtin_dict_type.html

I didn't really analyze if this one would hit any of `dict`'s pitfalls, but mostly just avoid it in general.

Reply:
Sure, that contains a good point, but there are reasons to do either. The implementation we have here isn't saved by inheriting from MutableMapping, however --  if anything, it perpetrates the same, by storing one thing in the dict and returning something else _just_ on `__getitem__`. It's a question of what most simply provides the expected interface.

Reply:
More to the point I was trying to get at – is there a reason to lazily convert values to `FeatureNameList`? At the extreme of simplicity, could this not be a function that returns a `dict` of `FeatureNameList`s keyed by table names?